### PR TITLE
Harsh/add holiday available time

### LINKED ISSUE #61: Feature - Add Time,Holiday and weekOff UI
**Is your feature request related to a problem? Please describe.**

Label - avalibleTime

- Right side addTime button
- Click on addTime button open time select time and selected time show on rounded textview background.
- Same way user add multiple times
- when click on update all time store in array format.

Label - Holidays.
- Right side addHolidays button
- Click on it open calendar select date
- Same way user add multiple holiday date
- when click on update all dates store in array format.

Label - WeekOff
- show 7 days check box.
- User selected checkbox value string in string []array.

DB:
timing: object[]
holidays: date[]
weekOff: string[]

**Add your acceptation criteria.**
Describe all user acceptation criteria in detail

### Check List

- [x] I have linked my PR with this task
- [x] I have added comment after completing this task
- [x] I have followed all [guidelines](https://github.com/jaykishan-sewak/doctor-appoint-app/tree/main) for code

### PR DESCRIPTION
**Pull request by HarshPanchalInfusion**
Add Holiday, week off, time slot PR

### CODE REVIEW

File: app/src/main/java/com/android/doctorapp/ui/doctor/AddDoctorViewModel.kt [hunk 1]
```diff
@@ -179,14 +197,13 @@ class AddDoctorViewModel @Inject constructor(
...
         if (isDoctor.value!!) {
-
             isUpdateDataValid.value = (!name.value.isNullOrEmpty() && !email.value.isNullOrEmpty()
                     && !contactNumber.value.isNullOrEmpty() && nameError.value.isNullOrEmpty()
                     && emailError.value.isNullOrEmpty() && contactNumberError.value.isNullOrEmpty()
                     && !address.value.isNullOrEmpty() && addressError.value.isNullOrEmpty()
                     && !dob.value.isNullOrEmpty() && dobError.value.isNullOrEmpty()
-                    && !isAvailableDate.value.isNullOrEmpty() && isAvailableDateError.value.isNullOrEmpty()
-                    && !availableTime.value.isNullOrEmpty() && availableTimeError.value.isNullOrEmpty()
+                    && availableTimeList.value?.isEmpty() == false && availableTimeError.value.isNullOrEmpty()
```
Comment (vikas-infusion) [inline app/src/main/java/com/android/doctorapp/ui/doctor/AddDoctorViewModel.kt]: availableTimeList.value?.isEmpty() == false you can use like this way also

availableTimeList.value.isNotEmpty()

File: app/src/main/java/com/android/doctorapp/ui/doctor/AddDoctorViewModel.kt [hunk 2]
```diff
@@ -375,6 +388,20 @@ class AddDoctorViewModel @Inject constructor(
             var recordId: String = ""
             session.getString(USER_ID).collectLatest {
                 val userData: UserDataRequestModel
+
+/*                weekDayList.forEachIndexed { index, weekOffModel ->
```
Comment (vikas-infusion) [inline app/src/main/java/com/android/doctorapp/ui/doctor/AddDoctorViewModel.kt]: Remove commented code

**Review by vikas-infusion — DISMISSED** [review]

File: app/src/main/java/com/android/doctorapp/ui/doctor/AddDoctorViewModel.kt (hunk 2 above)
Comment (HarshPanchalInfusion) [reply app/src/main/java/com/android/doctorapp/ui/doctor/AddDoctorViewModel.kt]: Done

File: app/src/main/java/com/android/doctorapp/ui/doctor/AddDoctorViewModel.kt (hunk 1 above)
Comment (HarshPanchalInfusion) [reply app/src/main/java/com/android/doctorapp/ui/doctor/AddDoctorViewModel.kt]: Done

**Review by vikas-infusion — DISMISSED** [review]

**Review by vikas-infusion — APPROVED** [review]

**Pull request merged**